subworkflows/local/bgc.nf

```diff
@@ -0,0 +1,100 @@
+/*
+    Run AMP screening tools
+*/
+include { UNTAR as UNTAR_CSS                       } from '../../modules/nf-core/modules/untar/main'
+include { UNTAR as UNTAR_DETECTION                 } from '../../modules/nf-core/modules/untar/main'
+include { UNTAR as UNTAR_MODULES                   } from '../../modules/nf-core/modules/untar/main'
+include { ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES } from '../../modules/nf-core/modules/antismash/antismashlitedownloaddatabases/main'
+include { ANTISMASH_ANTISMASHLITE                  } from '../../modules/nf-core/modules/antismash/antismashlite/main'
+
+workflow BGC {
+
+    take:
+    fna     // tuple val(meta), path(PROKKA.out.fna)
+    gff     // tuple val(meta), path(PROKKA.out.gff)
+
+    main:
+    ch_versions = Channel.empty()
+    ch_mqc      = Channel.empty()
+
+    // Check whether user supplies database and/or antismash directory. If not, obtain them via nf-core/modules/antismashlite/antismashlitedownloaddatabases.
+    if ( params.bgc_antismash_databases && params.bgc_antismash_directory ) {
+
+        // Supply databases
+        ch_antismash_databases = Channel
+            .fromPath( params.bgc_antismash_databases )
+            .first()
+
+        // Supply antiSMASH directory
+        ch_antismash_directory = Channel
+            .fromPath( params.bgc_antismash_directory )
+            .first()
+
+    } else if ( params.bgc_antismash_databases && !params.bgc_antismash_directory ) {
+
+        // Supply databases
+        ch_antismash_databases = Channel
+            .fromPath( params.bgc_antismash_databases )
+            .first()
+
+        // Supply antiSMASH directory
+        ch_css_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/css.tar.gz"
+        ch_detection_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/detection.tar.gz"
+        ch_modules_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/modules.tar.gz"
```

Comment:
You must not hardcode locations of objects like this into the pipeline, as it will prevent people from running the pipeline on clusters with compute-nodes that are not attached to the internet.

It might be safer to just not allow this combination, and force people to allways supply the directory.

Reply:
Decided to keep it in, with proper warnings/suggestion to use own antismash installation.